pkg/sources/sources.go

```diff
@@ -40,6 +40,8 @@ type Chunk struct {
 	SourceMetadata *source_metadatapb.MetaData
 	// SourceType is the type of Source that produced the chunk.
 	SourceType sourcespb.SourceType
+	// HandleMetadata holds the metadata from a handler if one was used.
+	HandleMetadata map[string]string
```

Comment:
I like this being part of the chunk.

Out of curiosity, do we have a formal definition of what a handler is? Does it take a chunk and produce multiple chunks, or is it 1:1?

Reply:
From what I understand, the implementation is: input `reader` -> create as many chunks as needed and then scan for secrets.